### ISSUE TITLE
`poetry run poetry ...` fails on Windows

### ISSUE THREAD

**Issue by johnthagen** [python-poetry/poetry#5909]
- [ ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10
- **Poetry version**: 1.1.13

## Issue

On Windows, it seems that the path to `poetry` itself is not accessible within a `poetry run` invocation from Powershell:

```bat
> poetry run poetry --version

  FileNotFoundError

  [WinError 2] The system cannot find the file specified

  at C:\Program Files\Python310\lib\subprocess.py:1435 in _execute_child
      1431│             sys.audit("subprocess.Popen", executable, args, cwd, env)
      1432│
      1433│             # Start the process
      1434│             try:
    → 1435│                 hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
      1436│                                          # no special security
      1437│                                          None, None,
      1438│                                          int(not close_fds),
      1439│                                          creationflags,
```

This did not fail for me with the same version of Poetry on macOS.

**Comment by dimbleby** [comment]
why would you do `poetry run poetry --version` and not just `poetry --version`?

**Comment by johnthagen** [comment]
@dimbleby This was a minimal reproducible example while trying to help debug another issue in which a program was calling `poetry` through `subprocess` and it failed on Windows. It seemed like it could be related.

**Comment by dimbleby** [comment]
fix at #3339 looks potentially relevant

**Comment by johnthagen** [comment]
On a clean Windows 10 VM, I installed Poetry 1.2.0b2:

```ps
(Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | py - --preview
```

And this then works:

```ps
> poetry run poetry --version
Poetry (version 1.2.0b2)
```

Closing this as fixed on latest pre-release.
